src/GraphQL/Execution/ParallelExecutionStrategy.cs

```diff
@@ -11,37 +11,36 @@ protected override Task ExecuteNodeTreeAsync(ExecutionContext context, ObjectExe
 
         protected async Task ExecuteNodeTreeAsync(ExecutionContext context, ExecutionNode rootNode)
         {
-            var pendingNodes = new List<ExecutionNode>
-            {
-                rootNode
-            };
+            var pendingNodes = new Queue<ExecutionNode>();
+            pendingNodes.Enqueue(rootNode);
 
+            var currentTasks = new List<Task<ExecutionNode>>();
             while (pendingNodes.Count > 0)
             {
-                var currentTasks = new Task<ExecutionNode>[pendingNodes.Count];
-
-                // Start executing all pending nodes
-                for (int i = 0; i < pendingNodes.Count; i++)
+                // Start executing pending nodes, while limiting the maximum number of parallel executed nodes to the set limit
+                while ((context.MaxParallelExecutionCount == null || currentTasks.Count < context.MaxParallelExecutionCount)
```

Comment:
it can now be argued that the behavior of `SerialExecutionStrategy` is equivalent to that of `ParallelExecutionStrategy` with `MaxParallelExecutionCount = 1`